### ISSUE TITLE
Toggle of BHoM popups on component outputs

### ISSUE THREAD

**Issue by Sonderwoods** [BHoM/Grasshopper_Toolkit#593]
I would love to have an option to disable the BHoM popups on the component outputs.
Right now I'm only using BHoM on selected projects, but my GH UI is messed up with all the BHoM popups.
Could we add a global toggle to this somehow?

![image](https://user-images.githubusercontent.com/19936679/106132362-26c4fd00-6164-11eb-920a-867bf4c6a580.png)

**Comment by Sonderwoods** [comment]
Either that or only enable the drop-down while the shift key is held, etc.

tir. 2. feb. 2021 20.00 skrev Al Fisher <notifications@github.com>:

**Comment by al-fisher** [comment]
> Either that or only enable the drop-down while the shift key is held, etc.

yes. This 👆

**Comment by hamedseifi** [comment]
"Right click on the bubbles" is not taken yet. It would be good to change the pop up feature of BHOM from left click on the bubbles to right click.

**Comment by DharmanGersch** [comment]
Yes please! I'd suggest <kbd>RMB</kbd> as long as it's only **near** the component param (not on the component, or the canvas because there are already useful menus there).

Most other options are already taken by very useful shortcuts. eg:

- <kbd>Shift</kbd>+ <kbd>LMB</kbd> = add wire
-  <kbd>Ctrl</kbd> +<kbd>Alt</kbd> = locate component
- etc https://www.grasshopper3d.com/forum/topics/what-hotkeys-and-shortcuts-are-available-in-grasshopper

It's really nice being able to immediately find what the input should be to a particular component param, 
![the good](https://user-images.githubusercontent.com/10596012/122497162-4a63e380-d030-11eb-8798-fce6293f2347.gif)

But it's frustrating not being able to create relays from double clicking params anymore 
With BHoM
![593 no OOTB no BHoM](https://user-images.githubusercontent.com/10596012/122496016-6cf4fd00-d02e-11eb-8d6d-cf4c3ccd699b.gif)

Without BHoM
![593 with BHoM](https://user-images.githubusercontent.com/10596012/122496786-abd78280-d02f-11eb-9393-8b95599946df.gif)

Maybe if the current behavior didn't block <kbd>LMB</kbd>+<kbd>LMB</kbd> from creating relays  <kbd>LMB</kbd> would be ok?

**Comment by adecler** [comment]
We can start by having a config file in our `C:\ProgramData\BHoM\Settings` folder that would control the behaviour of the pop-up menu. The possible options could be something like:
- Always on
- Only with Shift Key
- Only on scripts containing BHoM components
- Always off

The default option could be `Only on scripts containing BHoM components` as it seems to directly address the issue for people only using the BHoM in rare cases. 

The platform for setting files is already in place, including the ability to edit them from the UI so it mainly a question of using this new setting file in our menu controller:
![image](https://user-images.githubusercontent.com/16853390/122497421-e6392380-d01f-11eb-9bc5-26e97150cc31.png)

@hamedseifi , by `bubble` do you mean `input connector` ? This could be an option added to the list above to have the menu activate on right click instead of left but the right click on connectors is not a default behaviour of Grasshopper (that results to panning right now) so it depends on how easy it is to support that.

I am currently flat out on other projects but can have a look at this new config file once I have a bit of time.

**Comment by hamedseifi** [comment]
@adecler, Yes. I mean input and output connectors. 
![image](https://user-images.githubusercontent.com/86088258/122520684-44cdc400-d057-11eb-9a89-d2712b03783d.png)
